auth/postgres/groups.go

```diff
@@ -13,471 +13,493 @@ import (
-		mq = fmt.Sprintf("AND %s", mq)
+		cq = fmt.Sprintf(" %s WHERE %s", cq, mq)
+		mq = fmt.Sprintf(" AND %s", mq)
 	}
 
 	q := fmt.Sprintf(`SELECT id, owner_id, parent_id, name, description, metadata, path, nlevel(path) as level, created_at, updated_at FROM groups 
 					  WHERE nlevel(path) <= :level %s ORDER BY path`, mq)
-	cq := fmt.Sprintf("SELECT COUNT(*) FROM groups WHERE nlevel(path) <= :level %s", mq)
 
-	dbPage, err := toDBGroupPage("", "", "", "", level, gm)
+	dbPage, err := toDBGroupPage("", "", pm)
 	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
+		return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveAll, err)
 	}
 
 	rows, err := gr.db.NamedQueryContext(ctx, q, dbPage)
 	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
+		return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveAll, err)
 	}
 	defer rows.Close()
 
-	items, err := processRows(rows)
+	items, err := gr.processRows(rows)
 	if err != nil {
-		return groups.GroupPage{}, err
+		return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveAll, err)
 	}
 
 	total, err := total(ctx, gr.db, cq, dbPage)
 	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
+		return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveAll, err)
 	}
 
-	page := groups.GroupPage{
+	page := auth.GroupPage{
 		Groups: items,
-		PageMetadata: groups.PageMetadata{
+		PageMetadata: auth.PageMetadata{
 			Total: total,
+			Size:  uint64(len(items)),
 		},
 	}
 
 	return page, nil
 }
 
-func (gr groupRepository) RetrieveAllParents(ctx context.Context, groupID string, level uint64, gm groups.Metadata) (groups.GroupPage, error) {
-	if groupID == "" {
-		return groups.GroupPage{}, nil
-	}
-
-	_, mq, err := getGroupsMetadataQuery("groups", gm)
-	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errRetrieveDB, err)
-	}
-	if mq != "" {
-		mq = fmt.Sprintf("AND %s", mq)
-	}
-
-	q := fmt.Sprintf(`SELECT g.id, g.name, g.owner_id, g.parent_id, g.description, g.metadata, g.path, nlevel(g.path) as level, g.created_at, g.updated_at
-					  FROM groups parent, groups g
-					  WHERE parent.id = :parent_id AND g.path @> parent.path AND nlevel(parent.path) - nlevel(g.path) <= :level %s`, mq)
-
-	cq := fmt.Sprintf(`SELECT COUNT(*) FROM groups parent, groups g WHERE parent.id = :parent_id AND g.path @> parent.path %s`, mq)
-
-	if level > maxLevel {
-		level = maxLevel
-	}
-
-	dbPage, err := toDBGroupPage("", "", groupID, "", level, gm)
-	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
-	}
+func (gr groupRepository) RetrieveAllParents(ctx context.Context, groupID string, pm auth.PageMetadata) (auth.GroupPage, error) {
+	q := `SELECT g.id, g.name, g.owner_id, g.parent_id, g.description, g.metadata, g.path, nlevel(g.path) as level, g.created_at, g.updated_at
+		  FROM groups parent, groups g
+	      WHERE parent.id = :id AND g.path @> parent.path AND nlevel(parent.path) - nlevel(g.path) <= :level `
```

Comment:
Please remove the trailing whitespace. The same for the next line and all the other trailing whitespaces in the query strings.